StoreApp/Controllers/AccountController.cs

```diff
@@ -70,19 +88,194 @@ public ActionResult Login(UserLoginViewModel model)
+        {
+            UserEditViewModel user = webMapper.config.Map<UserDTO, UserEditViewModel>(userService.Get(id));
+
+            if (user != null && user.UserName.Equals(User.Identity.Name))
+            {
+                return View(user);
+            }
+
+            return RedirectToAction("Account");
+        }
+
+        [HttpPost]
+        public ActionResult EditUser(UserEditViewModel userEdit)
+        {
+            if (!ModelState.IsValid)
+            {
+                ModelState.AddModelError("", "Your edit account wrong!");
+                return View(userEdit);
+            }
+
+            UserDTO userDTO = webMapper.config.Map<UserEditViewModel, UserDTO>(userEdit);
+
+            if (!userValidateService.CheckTruePassword(userDTO))
+            {
+                ModelState.AddModelError("", "Your enter wrong password!");
+                return View(userEdit);
+            }
+
+            if (userValidateService.CheckForEditUser(userDTO))
+            {
+                userService.Edit(userDTO);
+
+                TempData["Message"] = "Your edit account successful!";
+
+                return RedirectToAction("Account");
+            }
+            else
+            {
+                ModelState.AddModelError("", "This is Email or Username already exist!");
+                return View(userEdit);
             }
         }
+
+        [HttpGet]
+        [Authorize]
+        public ActionResult DetailsUser(int id)
+        {
+            if (Session["Role"] as string == "admin")
+            {
+                var user = webMapper.config.Map<UserDTO, UserViewModel>(userService.Get(id));
+                return View(user);
+            }
+
+            return RedirectToAction("Account", "Account", null);
+        }
+
+        [HttpGet]
+        [Authorize]
+        public ActionResult EditUserAdmin(int id)
+        {
+            if (Session["Role"] as string == "admin")
+            {
+                UserViewModel user = webMapper.config.Map<UserDTO, UserViewModel>(userService.Get(id));
+
+                if (user != null)
+                {
+                    return View(user);
+                }
+
+                return RedirectToAction("Index");
+            }
+
+            return RedirectToAction("Account", "Account", null);
+        }
+
+        [HttpPost]
+        [Authorize]
+        public ActionResult EditUserAdmin(UserViewModel user)
+        {
+            if (Session["Role"] as string == "admin")
+            {
+                if (user.Role.Equals("0"))
+                {
+                    user.Role = "user";
+                }
+                else if (user.Role.Equals("1"))
+                {
+                    user.Role = "admin";
+                }
+
+                UserDTO userDTO = webMapper.config.Map<UserViewModel, UserDTO>(user);
+
+                userService.Edit(userDTO);
+
+                TempData["Message"] = user.UserName + " edit success";
```

Comment:
I propose to use this $"" to combine strings.
Also you should write readable message)

Reply:
fixed